Resources/doc/reference/filter_field_definition.rst

```diff
@@ -179,3 +179,31 @@ implement this functionality.
             return true;
         }
     }
+
+Filtering case insensitive strings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Normally the query builder behind the filtering would compare field values case sensitive. There can be reasons to
+filter case insensitive. To do so you can use the ``compare_case_insensitive`` option to force a case insensitive
+comparison on field value and filter value.
```

Comment:
maybe like this?

The default behaviour when filtering is to compare values in a case sensitive manner. For example "Test" is not the same as "test". Depending on your use case, you might want case insensitive filtering. To make the filter case insensitive, use the ``compare_case_insensitive`` option for the string filter: